front/src/modules/timetableItem/components/Timetable/RoundTrips/TodoColumn.tsx

```diff
@@ -73,6 +73,65 @@ const TodoColumn = ({
     setItemIdToPair(itemId);
   };
 
+  const pairItems = (candidateItem: PairingItem) => {
+    if (!itemIdToPair || !setPairingItems) return;
+    // This item is always in todo
+    const matchingItem = pairingItemsById?.get(itemIdToPair);
+    const timetableItemToPair = timetableItemsWithOpsById.get(itemIdToPair);
+    const candidateItemToPair = timetableItemsWithOpsById.get(candidateItem.id);
+    const candidateTimetableItem = timetableItemsWithOpsById.get(candidateItem.id);
+
+    if (!matchingItem || !timetableItemToPair || !candidateItemToPair || !candidateTimetableItem) {
+      throw new Error('Item to pair not found in todo column');
+    }
+
+    const isValidPair = checkRoundTripCompatible(timetableItemToPair, candidateTimetableItem);
+
+    setPairingItems((prevData) => {
+      const formattedMatchingItem: PairingItem = {
+        ...matchingItem,
+        status: 'roundTrips' as const,
+        pairedItemId: candidateItem.id,
+        isValidPair,
+      };
+
+      const formattedCandidateItem: PairingItem = {
+        ...candidateItem,
+        status: 'roundTrips' as const,
+        pairedItemId: itemIdToPair,
+        isValidPair,
+      };
+
+      // If the candidate item is already paired, we need to move its old pair to todo column
+      let formattedOrphanItem: PairingItem | undefined;
```

Comment:
nit: you can rename orphanItem -> orphanItemProps and then formattedOrphanItem->orphanItem
```suggestion
let orphanItem: PairingItem | undefined;
```

Reply:
Done